### ISSUE TITLE
MobileDateTimePicker does not render UTC in time picker if UTC is selected.

### ISSUE THREAD

**Issue by rdecoito** [mui/mui-x#12565]
### Search keywords

utc, mobiledatetimepicker, datetimepicker

### Latest version

- [X] I have tested the latest version

### Steps to reproduce

Link to live example: https://github.com/rdecoito/mui-datetime-utc-repro

Follow the README instructions (`npm i` and then `npm run dev`).

### Current behavior

When passing the `timezone='UTC'` prop to the MobileDateTimePicker and following the setup instructions for providing the Dayjs adapter, the time displayed in the upper right of the timepicker popup is displayed in the browser's local time.

Below is a screenshot of the field upon first render, which **correctly** displays the time in UTC (note: the time at creation of this image was 10:33 AM EST).

![image](https://github.com/mui/mui-x/assets/42098177/c3dcb570-93f1-4ab7-9610-c0550b41a9e7)

Now, look at the popup that appears after I click the field:

![image](https://github.com/mui/mui-x/assets/42098177/724054d1-04c8-48ed-8f20-ba9aa4abc3df)

You'll notice that the time is **incorrectly** displayed as 10:33 - which is EST - rather than 14:33 (UTC).

### Expected behavior

The time in the upper right of the timepicker popup should display in the timezone that was passed as a prop to the MobileDateTimePicker.

**Comment by michelengelen** [comment]
Thanks @rdecoito I can confirm this bug.
Additionally I noticed that the timepicker shows the correct hour, as shown in this screenshot
<img width="416" alt="Screenshot 2024-03-27 at 10 01 45" src="https://github.com/mui/mui-x/assets/32863416/d401ca61-2d38-4e42-87fc-a84a67b1238e">

@LukasTy this seems to be another problem with DayJs, since it does work as expected with Luxon:
<img width="422" alt="Screenshot 2024-03-27 at 10 07 00" src="https://github.com/mui/mui-x/assets/32863416/74c7a45d-70c6-469c-9d96-379f667567f1">

**Comment by rdecoito** [comment]
FYi, I did a bit of digging in the mui-x source code yesterday before making this issue and found the following:

You can see in the [DateTimePickerToolbar source code](https://github.com/mui/mui-x/blob/2ac14fd98fabe93663297512e2b219d329b384d9/packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx#L346) that the value you see in the upper right is formatted with `formatHours(value)` which [expands to](https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/DateTimePicker/DateTimePickerToolbar.tsx#L287) `utils.format(time, 'hours24h')` which in turn [expands to](https://github.com/mui/mui-x/blob/2ac14fd98fabe93663297512e2b219d329b384d9/packages/x-date-pickers/src/AdapterDayjs/AdapterDayjs.ts#L81) `dayjs(time).format('HH')`. This is as far as I got before I had to move on to other stuff at work.

That last bit there (`dayjs(time).format('HH')`) will always format in the browser's local time despite the `dayjs.extend(utc)` line. However, if I provide the same format string to the `format={}` prop of the MobileDateTimePicker, it gets formatted in UTC correctly. The code and output below demonstrate this discrepancy (you can see this in the [`dayjs-format-repro`](https://github.com/rdecoito/mui-datetime-utc-repro/tree/dayjs-format-repro) branch of my reproduction repo):

![image](https://github.com/mui/mui-x/assets/42098177/b2313beb-57a4-4a74-b809-a029f9e3a942) 
![image](https://github.com/mui/mui-x/assets/42098177/ca9518b8-a504-48af-901f-87684923bbdf)

That ^^ `09` is in EST and the `13` is UTC, despite the same formatting string being used. So clearly dayjs _can_ be used to format HH into UTC somehow.

Basically, my takeaway here (best guess as someone unfamiliar with the codebase) is that the datetimepicker is somehow _not_ providing localization info/context to the DateTimePickerToolbar's formatting logic, but _is_ providing that info/context to the formatting in the input field itself. So it seems plausible this is less of an intrinsic dayjs problem and more related to missing timezone logic in the wrapping logic of the DateTimePickerToolbar.

I'm not sure though. Maybe that was all useless, but I just wanted to throw these thoughts down in case it helps whoever tackles this problem.

**Comment by gtRfnkN** [comment]
This issue still exists. Was there any progress?

**Comment by LukasTy** [comment]
This issue looks like a duplicate that has already been resolved with https://github.com/mui/mui-x/pull/13481. 😉 👌 
The fix is available starting v7.7.1.
